collector/logical_disk.go

```diff
@@ -204,98 +202,98 @@ func (c *LogicalDiskCollector) collect(ch chan<- prometheus.Metric) (*prometheus
 		ch <- prometheus.MustNewConstMetric(
 			c.RequestsQueued,
 			prometheus.GaugeValue,
-			float64(volume.CurrentDiskQueueLength),
+			volume.CurrentDiskQueueLength,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.ReadBytesTotal,
 			prometheus.CounterValue,
-			float64(volume.DiskReadBytesPerSec),
+			volume.DiskReadBytesPerSec,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.ReadsTotal,
 			prometheus.CounterValue,
-			float64(volume.DiskReadsPerSec),
+			volume.DiskReadsPerSec,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.WriteBytesTotal,
 			prometheus.CounterValue,
-			float64(volume.DiskWriteBytesPerSec),
+			volume.DiskWriteBytesPerSec,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.WritesTotal,
 			prometheus.CounterValue,
-			float64(volume.DiskWritesPerSec),
+			volume.DiskWritesPerSec,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.ReadTime,
 			prometheus.CounterValue,
-			float64(volume.PercentDiskReadTime)*ticksToSecondsScaleFactor,
+			volume.PercentDiskReadTime,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.WriteTime,
 			prometheus.CounterValue,
-			float64(volume.PercentDiskWriteTime)*ticksToSecondsScaleFactor,
+			volume.PercentDiskWriteTime,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.FreeSpace,
 			prometheus.GaugeValue,
-			float64(volume.PercentFreeSpace)*1024*1024,
+			volume.PercentFreeSpace_Base*1024*1024,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
 			c.TotalSpace,
 			prometheus.GaugeValue,
-			float64(volume.PercentFreeSpace_Base)*1024*1024,
+			volume.PercentFreeSpace*1024*1024,
```

Comment:
You're swapping base/non-base here, is this intentional?

Reply:
Yes, this is deliberate. The change is confusing, but the new perflib `PercentFreeSpace_Base` matches with the WMI `PercentFreeSpace` and vice versa.
Same for the perflib `PercentFreeSpace` and WMI `PercentFreeSpace_Base`.

I had to re-review this section a number of times :confounded: but the output metric value is the same between the `0.8.3` WMI and the proposed perflib.

Reply:
Ok! :+1: